diff --git a/ios/ExpoLinkingRegistry.swift b/ios/ExpoLinkingRegistry.swift
index 98de34c6e22aaf652b834eb33f1772f439dd9708..eb5b580cb4a4dc141d26b61a67cf4a2d1f92f651 100644
--- a/ios/ExpoLinkingRegistry.swift
+++ b/ios/ExpoLinkingRegistry.swift
@@ -5,5 +5,5 @@ typealias OnURLReceivedCallback = (URL) -> Void
 public let clearInitialURLNotification = Notification.Name("ExpoLinkingClearInitialURL")
 
-class ExpoLinkingRegistry: NSObject {
+class ExpoLinkingRegistry: NSObject, @unchecked Sendable {
   static let shared = ExpoLinkingRegistry()
   var initialURL: URL?
