diff --git a/scripts/codegen/generate-artifacts-executor/generateRCTThirdPartyComponents.js b/scripts/codegen/generate-artifacts-executor/generateRCTThirdPartyComponents.js
index 11f0bbd74d0815839b37de0107dd6174ed4bd794..e4f7a4d2a0b9a9ec8b4b2d6a1d7e5f3a7a0a5c8f 100644
--- a/scripts/codegen/generate-artifacts-executor/generateRCTThirdPartyComponents.js
+++ b/scripts/codegen/generate-artifacts-executor/generateRCTThirdPartyComponents.js
@@ -130,7 +130,7 @@ function generateRCTThirdPartyComponents(
     .flatMap(library => {
       const components = componentsInLibraries[library];
       return components.map(({componentName, className}) => {
-        return `\t\t@"${componentName}": NSClassFromString(@"${className}"), // ${library}`;
+        return `\t\tClass ${componentName}Class = NSClassFromString(@"${className}"); // ${library}\n\t\tif (${componentName}Class != Nil) {\n\t\t\tthirdPartyComponents[@"${componentName}"] = ${componentName}Class;\n\t\t}`;
       });
     })
     .join('\n');
diff --git a/scripts/codegen/templates/RCTThirdPartyComponentsProviderMM.template b/scripts/codegen/templates/RCTThirdPartyComponentsProviderMM.template
index 95115f0f6e96f049fa859b81843d304088434a7b..610d2a98733db19b2fd080a64a82084a1cc0fa2c 100644
--- a/scripts/codegen/templates/RCTThirdPartyComponentsProviderMM.template
+++ b/scripts/codegen/templates/RCTThirdPartyComponentsProviderMM.template
@@ -16,15 +16,13 @@
 + (NSDictionary<NSString *, Class<RCTComponentViewProtocol>> *)thirdPartyFabricComponents
 {
-  static NSDictionary<NSString *, Class<RCTComponentViewProtocol>> *thirdPartyComponents = nil;
+  static NSMutableDictionary<NSString *, Class<RCTComponentViewProtocol>> *thirdPartyComponents = nil;
   static dispatch_once_t nativeComponentsToken;
 
   dispatch_once(&nativeComponentsToken, ^{
-    thirdPartyComponents = @{
+    thirdPartyComponents = [NSMutableDictionary new];
 {thirdPartyComponentsMapping}
-    };
   });
 
   return thirdPartyComponents;
 }
 
